app/controlplane/pkg/biz/user.go

```diff
@@ -134,17 +151,30 @@ func (uc *UserUseCase) FindOrCreateByEmail(ctx context.Context, email string, di
 			LoggedIn: time.Now(),
 		}, nil)
 
+		if (opts.FirstName != nil && u.FirstName != *opts.FirstName) ||
+			(opts.LastName != nil && u.LastName != *opts.LastName) {
```

Comment:
Here @jiparis. This line ensures that if new first or last name information is provided and it differs from what is currently stored for the user, the user's name in the database will be updated accordingly.

By default the `FirstName` and `LastName` of all users after applying the migration will be NULL which on the struct is "", so yes, they won't get updated if they are the same as the incoming ones, only different.